azure-iot-device/tests/common/test_mqtt_transport.py

```diff
@@ -780,6 +883,67 @@ def test_client_returns_failing_rc_code(
         with pytest.raises(error_params["error"]):
             transport.disconnect()
 
+    @pytest.mark.it("Stops MQTT Network Loop when disconnect does not raise an exception")
+    def test_calls_loop_stop_on_success(self, mocker, mock_mqtt_client, transport):
+        transport.disconnect()
+
+        assert mock_mqtt_client.loop_stop.call_count == 1
+        assert mock_mqtt_client.loop_stop.call_args == mocker.call()
+
+    @pytest.mark.it("Stops MQTT Network Loop when disconnect raises an exception")
+    def test_calls_loop_stop_on_exception(
+        self, mocker, mock_mqtt_client, transport, arbitrary_exception
+    ):
+        mock_mqtt_client.disconnect.side_effect = arbitrary_exception
+
+        with pytest.raises(Exception):
+            transport.disconnect()
+
+        assert mock_mqtt_client.loop_stop.call_count == 1
+        assert mock_mqtt_client.loop_stop.call_args == mocker.call()
+
+    @pytest.mark.it(
+        "Sets Paho's _thread to None if disconnect does not raise an exception while running in the Paho thread"
+    )
+    def test_sets_thread_to_none_on_success_in_paho_thread(
+        self, mocker, mock_mqtt_client, transport, mock_paho_thread_current
+    ):
+        transport.disconnect()
+        assert mock_mqtt_client._thread is None
+
+    @pytest.mark.it(
+        "Sets Paho's _thread to None if disconnect raises an exception while running in the Paho thread"
```

Comment:
>_thread [](start = 21, length = 7)

I generally try to refer to things like this as a "._thread attribute" or something to indicate what the thing we're talking about is. Although depending on the test, it can get a bit too verbose to always do that.